Sources/Core/AWSSDKIdentity/Sources/AWSSDKIdentity/AWSCredentialIdentityResolvers/ProfileAWSCredentialIdentityResolver.swift

```diff
@@ -191,8 +208,16 @@ public struct ProfileAWSCredentialIdentityResolver: AWSCredentialIdentityResolve
                     source: credSource,
                     identityProperties: identityProperties
                 )
+                let sourceCredsWithUpdatedFeatureIDs = prependFeatureIDs(
+                    featureIDsToPrepend: credentialFeatureIDs + [
+                        CredentialFeatureID.CREDENTIALS_PROFILE_NAMED_PROVIDER.rawValue
+                    ],
+                    creds: sourceCreds
+                )
                 return try await STSAssumeRoleAWSCredentialIdentityResolver(
-                    awsCredentialIdentityResolver: StaticAWSCredentialIdentityResolver(sourceCreds),
+                    awsCredentialIdentityResolver: StaticAWSCredentialIdentityResolver(
+                        sourceCredsWithUpdatedFeatureIDs
+                    ),
                     roleArn: profile.string(for: .init(stringLiteral: "role_arn"))!,
                     sessionName: profile.string(for: .init(stringLiteral: "role_session_name"))
                 ).getIdentity(identityProperties: identityProperties)
```

Comment:
`credentialFeatureIDs`, i.e., the feature IDs resolved so far, does not need to be passed in to the STSAssumeRole credential resolver separately in order to include that in the assume role request because the credentials that will be provided by StaticAWSCredentialIdentityResolver already contains that value in its properties & `BusinessMetric` will fetch it from resolved identity object that contains the needed feature IDs.